lib/matplotlib/tests/test_axes.py

```diff
@@ -1744,6 +1744,120 @@ def test_hist_zorder(histtype, zorder):
+    ref_axes[3].set_xlim(0, None)
+
+    ref_axes[4].plot(x, np.append(y, y[-1]), drawstyle='steps-post')
+    ref_axes[4].add_line(mlines.Line2D([x[0], x[0]], [0, y[0]]))
+    ref_axes[4].add_line(mlines.Line2D([x[-1], x[-1]], [0, y[-1]]))
+    ref_axes[4].semilogy()
+
+    ref_axes[5].plot(np.append(y[0], y), x, drawstyle='steps-post')
+    ref_axes[5].add_line(mlines.Line2D([0, y[0]], [x[0], x[0]]))
+    ref_axes[5].add_line(mlines.Line2D([0, y[-1]], [x[-1], x[-1]]))
+    ref_axes[5].semilogx()
+
+
+@check_figures_equal()
+def test_stairs_fill(fig_test, fig_ref):
+    h, bins = [1, 2, 3, 4, 2], [0, 1, 2, 3, 4, 5]
+    bs = -2
+    # Test
+    fig_test, test_axes = plt.subplots(2, 2)
+    test_axes = test_axes.flatten()
+    test_axes[0].stairs(h, bins, fill=True)
+    test_axes[1].stairs(h, bins, orientation='horizontal', fill=True)
+    test_axes[2].stairs(h, bins, baseline=bs, fill=True)
+    test_axes[3].stairs(h, bins, baseline=bs, orientation='horizontal',
+                          fill=True)
+
+    # # Ref
+    fig_ref, ref_axes = plt.subplots(2, 2)
+    ref_axes = ref_axes.flatten()
+    ref_axes[0].fill_between(bins, np.append(h, h[-1]), step='post')
+    ref_axes[0].set_ylim(0, None)
+    ref_axes[1].fill_betweenx(bins, np.append(h, h[-1]), step='post')
+    ref_axes[1].set_xlim(0, None)
+    ref_axes[2].fill_between(bins, np.append(h, h[-1]),
+                             np.ones(len(h)+1)*bs, step='post')
+    ref_axes[2].set_ylim(bs, None)
+    ref_axes[3].fill_betweenx(bins, np.append(h, h[-1]),
+                              np.ones(len(h)+1)*bs, step='post')
+    ref_axes[3].set_xlim(bs, None)
+
+
+@check_figures_equal()
+def test_stairs_update(fig_test, fig_ref):
+    # Test
+    fig_test, test_ax = plt.subplots()
+    h = test_ax.stairs([1, 2])
+    h.set_values([2, 1])
+    h.set_edges([0, 1, 1.5])
+
+    # # Ref
+    fig_ref, ref_ax = plt.subplots()
+    h = ref_ax.stairs([2, 1], [0, 1, 1.5])
+
+
+@pytest.mark.xfail
+def test_stairs_invalid_nan():
+    plt.stairs([1, 2], [0, np.nan, 1])
+
+
+@pytest.mark.xfail
+def test_stairs_invalid_mismatch():
+    plt.stairs([1, 2], [0, 1])
+
+
+@image_comparison(['test_stairs_options.png'], remove_text=True)
+def test_stairs_options():
+    x, y = np.array([1, 2, 3, 4, 5]), np.array([1, 2, 3, 4]).astype(float)
+    yn = y.copy()
+    yn[1] = np.nan
+
+    fig, ax = plt.subplots()
+    ax.stairs(y*3, x, color='green', fill=True, label="A")
+    ax.stairs(y, x*3-3, color='red', fill=True,
+              orientation='horizontal', label="B")
+    ax.stairs(yn, x, color='orange', ls='--', lw=2, label="C")
+    ax.stairs(yn/3, x*3-2, ls='--', lw=2, baseline=0.5,
+              orientation='horizontal', label="D")
+    ax.stairs(y[::-1]*3+12, x, color='red', ls='--', lw=2, baseline=None,
+              label="E")
+    ax.stairs(y[:-1][::-1]*2+11, x[:-1]+0.5, color='black', ls='--', lw=2,
+              baseline=12, hatch='//', label="F")
+    ax.legend(loc=0)
+
+
```

Comment:
Can we add a datetime test?  i/e. `edges=np.arange('2001-01-01', '2002-01-10')`?

Reply:
![image](https://user-images.githubusercontent.com/13226500/92593309-2f926080-f2a1-11ea-8b6d-50f048bf3c11.png)

Idk if I like this interpolating behaviour for the ticks.

Reply:
What interpolation behaviour?  The extra ticks?  You'd get the same thing for integers wouldn't you?  The tick locator doesn't know it has a "histogram" being plotted above the axis.

Reply:
Hmm, yeah you're right. Thoughts on adding that as a feature? Not in this PR though.

Reply:
There are lots of ways of setting the ticks.  

Anyhow, it would be nice to have a test of datetime because it tests that axis-units are handled...

Reply:
Whoops, seems I added the units handling, but forgot the test.